File: Simple Linear Regression/simple_linear_regression.py
# Simple linear regression

# Import libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Import dataset
dataset = pd.read_csv('Salary_Data.csv')

X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Split into training and test sets
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Simple linear regression on the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Prediction from the test set
y_pred = regressor.predict(X_test)

# Plot the training set results
plt.scatter(X_train, y_train, color = "red")
plt.plot(X_train, regressor.predict(X_train), color = "blue")
plt.title("Salary vs. experience (Training set)")
plt.xlabel("Years of experience")
plt.ylabel("Salary in dollars")
plt.show()

# Plot the test set results
plt.scatter(X_test, y_test, color = "red")
plt.plot(X_train, regressor.predict(X_train), color = "blue")
plt.title("Salary vs. experience (Test set)")
plt.xlabel("Years of experience")
plt.ylabel("Salary in dollars")
plt.show()

File: Simple Linear Regression/simple_linear_regression.R
# Loading the dataset
dataset <- read.csv("Salary_Data.csv")


# Split into training and test set
library(caTools)
set.seed(123)
split <- sample.split(dataset$Salary, SplitRatio = 2/3)
training_set <- dataset[split,]
test_set <- dataset[!split,]

# Fit model
regressor <- lm(Salary ~ YearsExperience, data = training_set)

# Predict test set results
y_pred <- predict(regressor, test_set)

# Visualize the training set results
library(ggplot2)
ggplot() +
  geom_point(aes(x = YearsExperience, y = Salary), 
             col = "red", data = training_set) +
  geom_line(aes(x = training_set$YearsExperience,
                y = predict(regressor, training_set)), 
            col = "blue") +
  ggtitle("Salary vs. experience (Training set)") +
  xlab("Years of experience") +
  ylab("Salary in dollars")

# Visualize the test set results
ggplot() +
  geom_point(aes(x = YearsExperience, y = Salary), 
             col = "red", data = test_set) +
  geom_line(aes(x = training_set$YearsExperience,
                y = predict(regressor, training_set)), 
            col = "blue") +
  ggtitle("Salary vs. experience (Test set)") +
  xlab("Years of experience") +
  ylab("Salary in dollars")
